### PR TITLE
implement `serde::Deserialize` for Value

### PR DESCRIPTION
**Pull request by euclio**
Progress towards #43.

### CODE REVIEW

**Comment by euclio** [comment]
I can't reproduce the build failure locally. Probably fixed in the latest nightly?

**Comment by 3Hren** [comment]
Wow! Thanks for the contribution!

I think that successful build on nightly is not required - will fix travis script soon.

File: rmp/src/value.rs [hunk 1]
```diff
@@ -481,31 +481,51 @@ impl From<usize> for Value {
 
 impl From<i8> for Value {
     fn from(v: i8) -> Value {
-        Value::Integer(Integer::I64(From::from(v)))
```
Comment (3Hren) [inline rmp/src/value.rs]: Why have you changed this? I think the proper solution will be to check whether the incoming value fits in the shortest-bytes-representation requirement.

**Comment by 3Hren** [comment]
Everything is ok, except one place I marked with my comment. Let's discuss and I'll merge.

File: rmp/src/value.rs (hunk 1 above)
Comment (euclio) [inline rmp/src/value.rs]: I changed it due to the comment on https://github.com/3Hren/msgpack-rust/blob/master/rmp/src/value.rs#L14, and makes `Value::from` the same as `Deserialize`. I don't feel strongly either way, but it does simplify the tests.

**Pull request merged**